src/types/store/tablets.ts

```diff
@@ -0,0 +1,42 @@
+import type {ApiRequestAction} from '../../store/utils';
+import {
+    clearWasLoadingFlag,
+    FETCH_TABLETS,
+    setStateFilter,
+    setTypeFilter,
+} from '../../store/reducers/tablets';
+
+import type {IResponseError} from '../api/error';
+import type {TEvTabletStateResponse, EType, ETabletState} from '../api/tablet';
+
+export interface ITabletsState {
+    loading: boolean;
+    wasLoaded: boolean;
+    stateFilter: ETabletState[];
+    typeFilter: EType[];
+    data?: TEvTabletStateResponse;
+    error?: IResponseError;
+}
+
+export interface ITabletsApiRequestParams {
+    nodes?: string[];
+    path?: string;
+}
+
+type ITabletsApiRequestAction = ApiRequestAction<
+    typeof FETCH_TABLETS,
+    TEvTabletStateResponse,
+    IResponseError
+>;
+
+export type ITabletsAction =
+    | ITabletsApiRequestAction
+    | (
+          | ReturnType<typeof clearWasLoadingFlag>
+          | ReturnType<typeof setStateFilter>
+          | ReturnType<typeof setTypeFilter>
+      );
```

Comment:
why group the ReturnTypes?

Reply:
No actual reason. I did this previously, it seems more beautiful for me